libs/framework/include/service_registration.h

```diff
@@ -39,9 +39,6 @@ FRAMEWORK_EXPORT celix_status_t serviceRegistration_unregister(service_registrat
 FRAMEWORK_EXPORT celix_status_t
 serviceRegistration_getProperties(service_registration_t *registration, celix_properties_t **properties);
 
-FRAMEWORK_EXPORT celix_status_t
-serviceRegistration_setProperties(service_registration_t *registration, celix_properties_t *properties);
```

Comment:
There are two problems with the current design:

1. memory leak in `serviceRegistration_setProperties`
2. 'serviceRegistration_getProperties' is inherently unsafe in multi-threading

Given that there is currently no user of setProperties, I suggest making properties read-only.

Reply:
Yeah I agree. 
Ideally we do not update the public headers (even the ones without a celix_ prefix) backwards incompatible. But if it not used in Celix, probably also not used by users and it is problematic then we should make an exception.

Side note: In the future I would like to make a 3.0 Celix release where all non celix_ prefix c headers are no longer installed and moved to the src dirs.

Reply:
The 3.0 move will be fantastic! At a moment I suddenly realized only headers included in celix_api.h should be considered public.